pkg/config/model/viper.go

```diff
@@ -89,13 +90,18 @@ func (c *safeConfig) Set(key string, value interface{}, source Source) {
 		return
 	}
 
-	c.Lock()
-	defer c.Unlock()
-	c.configSources[source].Set(key, value)
-	c.mergeViperInstances(key)
+	// modify the config then release the lock to avoid deadlocks
+	var receivers []NotificationReceiver
+	func() {
+		c.Lock()
+		defer c.Unlock()
+		c.configSources[source].Set(key, value)
+		c.mergeViperInstances(key)
+		receivers = slices.Clone(c.notificationReceivers)
+	}()
 
 	// notifying all receiver about the updated setting
-	for _, receiver := range c.notificationReceivers {
+	for _, receiver := range receivers {
```

Comment:
You have a race issue as `receivers` is set line `100` by another goroutine
What do you think about the following code:
```go
var receivers []NotificationReceiver
c.Lock()
c.configSources[source].Set(key, value)
c.mergeViperInstances(key)
receivers = slices.Clone(c.notificationReceivers)
c.Unlock()

for _, receiver := range receivers {
...
```

Reply:
Sure, that sounds good. It wasn't using a goroutine before, but regardless, I agree this looks better.